### PR TITLE
♻️ chore(slo): SLOs for CommitContextIntegration

### PR DESCRIPTION
**Pull request by iamrajjoshi**
take 2, added ecosystem slo event lifecycles to the mixin.

### CODE REVIEW

File: tests/sentry/integrations/source_code_management/test_commit_context_slo.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+from sentry.models.repository import Repository
+from sentry.testutils.cases import TestCase
+from sentry.users.models.identity import Identity
+from tests.sentry.integrations.utils.test_assert_metrics import (
+    assert_failure_metric,
+    assert_halt_metric,
+)
+
+
+class MockCommitContextIntegration(CommitContextIntegration):
```
Comment (iamrajjoshi) [inline tests/sentry/integrations/source_code_management/test_commit_context_slo.py:17]: this was the easiest way to test this.

File: tests/sentry/integrations/utils/test_assert_metrics.py [hunk 2]
```diff
@@ -6,14 +6,20 @@
 
 
 def assert_halt_metric(mock_record, error_msg):
-    (event_failures,) = (
+    (event_halts,) = (
         call for call in mock_record.mock_calls if call.args[0] == EventLifecycleOutcome.HALTED
     )
-    assert event_failures.args[1] == error_msg
+    if isinstance(error_msg, Exception):
```
Comment (iamrajjoshi) [inline tests/sentry/integrations/utils/test_assert_metrics.py:12]: this allows us to use these helpers when we pass an exception as well

File: tests/sentry/tasks/test_commit_context.py [hunk 3]
```diff
@@ -1223,10 +1229,18 @@ def test_gh_comment_debounces(self, get_jwt, mock_comment_workflow, mock_get_com
             )
             assert mock_comment_workflow.call_count == 1
 
+        start_1, failure_1, start_2, failure_2 = mock_record.mock_calls
+        assert start_1.args[0] == EventLifecycleOutcome.STARTED
+        assert failure_1.args[0] == EventLifecycleOutcome.SUCCESS
```
Comment (GabeVillalobos) [inline tests/sentry/tasks/test_commit_context.py:1234]: Minor nit: variable names for the failures here don't match the outcomes.

**Review by GabeVillalobos — APPROVED** [review]

File: src/sentry/integrations/source_code_management/commit_context.py [hunk 4]
```diff
@@ -94,16 +100,28 @@ def get_blame_for_files(
...
+            try:
+                client = self.get_client()
+            except Identity.DoesNotExist as e:
+                lifecycle.record_failure(e)
+                sentry_sdk.capture_exception(e)
+                return []
+            try:
+                response = client.get_blame_for_files(files, extra)
+            except IdentityNotValid as e:
+                lifecycle.record_failure(e)
```
Comment (cathteng) [inline src/sentry/integrations/source_code_management/commit_context.py:116]: should we ask Issues team why these specific errors are caught? i'm not sure if this is a failure on our side or the user's side. it seems like this happens when refreshing the identity fails

Comment (iamrajjoshi) [reply src/sentry/integrations/source_code_management/commit_context.py:116]: let me do that. i think we can record_failure to see how often this happens then adjust?

File: src/sentry/integrations/source_code_management/commit_context.py [hunk 5]
```diff
@@ -120,114 +138,137 @@ def queue_comment_task_if_needed(
...
+        ).capture() as lifecycle:
+            if not OrganizationOption.objects.get_value(
+                organization=project.organization,
+                key="sentry:github_pr_bot",
+                default=True,
+            ):
+                # TODO: remove logger in favor of the log recorded in  lifecycle.record_halt
+                logger.info(
+                    _pr_comment_log(integration_name=self.integration_name, suffix="disabled"),
+                    extra={"organization_id": project.organization_id},
```
Comment (cathteng) [inline src/sentry/integrations/source_code_management/commit_context.py:156]: i remember we added logging when halts happen. does this mean we can remove these logs?

**Pull request merged**